src/api_server/src/token/mod.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::*;
+use serde::Deserialize;
+use serde_json::Value;
+use std::sync::Arc;
+use strum_macros::EnumString;
+use tokio::sync::RwLock;
+
+mod simple;
+
+pub trait AttestationTokenBroker {
+    /// Issue an signed attestation token with custom claims.
+    /// Return base64 encoded Token.
+    fn issue(&self, custom_claims: Value) -> Result<String>;
+
+    /// Verify an signed attestation token
+    fn verify(&self, token: String) -> Result<()>;
```

Comment:
For `verify` to be useful to callers, it needs to return the claims from the token, and in particular the TEE pubkey (as that is what should be used to wrap secrets).